java/src/jmri/jmrit/XmlFileValidateAction.java

```diff
@@ -71,29 +71,28 @@ public void actionPerformed(ActionEvent e) {
     }
 
     protected void processFile(File file) {
-        if (log.isDebugEnabled()) {
-            log.debug("located file " + file + " for XML processing");
-        }
-        // handle the file (later should be outside this thread?)
+        log.debug("located file \"{}\" for XML processing", file);
+        // handle the file (TODO should be outside this thread?)
         try {
             xmlfile.setValidate(XmlFile.Validate.CheckDtdThenSchema);
             readFile(file);
         } catch (Exception ex) {
-            showFailResults(_who, ex.getMessage());
+            showFailResults(_who, file.getName(), ex.getMessage());
             return;
         }
-        showOkResults(_who, "OK");
-        if (log.isDebugEnabled()) {
-            log.debug("parsing complete");
-        }
+        showOkResults(_who, Bundle.getMessage("ValidatedOk", file.getName()));
+        log.debug("parsing xml complete");
     }
 
     protected void showOkResults(Component who, String text) {
         JOptionPane.showMessageDialog(who, text);
     }
 
-    protected void showFailResults(Component who, String text) {
-        JOptionPane.showMessageDialog(who, text);
+    protected void showFailResults(Component who, String fileName, String text) {
+        final String html = "<html><body style='width: %1spx'>%1s"; // reflow in dialog
+        final int dialogWidth = 300;
+        JOptionPane.showMessageDialog(who, String.format(html, dialogWidth,
+                Bundle.getMessage("ValidationErrorInFile", fileName) + "<br><br>" + text)); // html markup
```

Comment:
Instead of concatenating text to the end of the format string, it would likely be better to just include in the format string.

Reply:
Thanks, done (shortly)